wodles/aws/aws_s3.py

```diff
@@ -3325,7 +3306,189 @@ def purge_db(self, log_group):
+
+
+class AWSSSecLakeMessageProcessor(AWSQueueMessageProcessor):
+    def parse_message(self, message: dict) -> dict:
+        try:
+            log_path = message["detail"]["object"]["key"]
+            bucket_path = message["detail"]["bucket"]["name"]
+            return {"route": {"log_path": log_path, "bucket_path": bucket_path}}
+        except KeyError:
+            return {'raw_message': message}
+
+
+class AWSS3LogHandler:
+    def obtain_logs(self, bucket: str, log_path: str) -> list:
+        """
+        Fetch a file from a bucket and obtain a list of events from it.
+
+        Parameters
+        ----------
+        bucket : str
+            Bucket to get the file from.
+        log_path : str
+            Relative path of the file inside the bucket.
+
+        Returns
+        -------
+        list[dict]
+            List of extracted events to send to Wazuh.
+        """
+        raise NotImplementedError
+
+    def process_file(self, message_body: dict) -> None:
+        """Parse an SQS message body, obtain the events associated, and send them to Analysisd.
+
+        Parameters
+        ----------
+        message_body : dict
+            An SQS message received from the queue.
+        """
+        raise NotImplementedError
+
+
+class AWSSubscriberBucket(WazuhIntegration, AWSS3LogHandler):
+    """
+    Class for processing events from AWS S3 buckets.
+
+    Attributes
+    ----------
+    aws_profile : str
+        AWS profile.
+    iam_role_arn : str
+        IAM Role.
+    """
+    def __init__(self, service_endpoint: str = None, sts_endpoint: str = None, profile: str = None, **kwargs):
+        WazuhIntegration.__init__(self, access_key=None,
+                                  secret_key=None,
+                                  aws_profile=profile,
+                                  service_name='s3',
+                                  service_endpoint=service_endpoint,
+                                  sts_endpoint=sts_endpoint,
+                                  **kwargs)
+
+    def obtain_logs(self, bucket: str, log_path: str) -> list[dict]:
+        """
+        Fetch a file from a bucket and obtain a list of events from it.
+
+        Parameters
+        ----------
+        bucket : str
+            Bucket to get the file from.
+        log_path : str
+            Relative path of the file inside the bucket.
+
+        Returns
+        -------
+        list[dict]
+            List of extracted events to send to Wazuh.
+        """
+
+        def _process_jsonl(file):
+            json_list = list(file)
+            result = []
+            for json_item in json_list:
+                x = json.loads(json_item)
+                result.append(dict(x, integration='aws'))
+            return result
+
+        def json_event_generator(data):
+            while data:
+                json_data, json_index = decoder.raw_decode(data)
+                data = data[json_index:]
+                yield json_data
```

Comment:
I suggest moving these functions as private methods of the class, to improve the legibility.

Reply:
Defined as static method of the class